File: OOP-Java/output/com/output34/Vehicle.java
package com.output34;

public final class Vehicle {	// final class - we can't inherit final class.
	protected String color;
	private int maxSpeed;

	public Vehicle(int maxSpeed) {
		this.maxSpeed = maxSpeed;
		this.color = "Blue";
	}

	public void print() {		// final - we can't overide final function
		System.out.println("Vehicle's print function called !");
	}
}

File: OOP-C++/Part3/Basics/6 Polymorphism/Abstract Class/Vehicle.cpp
class Vehicle {

	public:
		string color;

		virtual void print() = 0;	// Pure Virtual Function
};

/*
   Any class having pure virtual function is called abstract class.
   Abstract means incomplete.
   We can't create object of abstract class.
 */

/*
   If any class inherit abstract class :
    1. Implement all the pure virtual functions.
    2. Become an abstract class.
 */

/*
   Pure virtual function is used when we want to enforce our child class to overide
   its function.
 */
